firstvoices/backend/serializers/site_data_serializers.py

```diff
@@ -38,86 +54,100 @@ def get_site_data_export(self, site):
             "L2": {"name": "English"},
             "build": datetime.now().strftime("%Y%m%d%H%M"),
         }
-        queryset = (
-            site.dictionaryentry_set
-            if site is not None and site.dictionaryentry_set is not None
-            else None
-        )
-        request = self.context.get("request")
-        dictionary_entries = DictionaryEntryDataSerializer(
-            queryset, many=True, context={"request": request}
-        ).data
-
-        return {"config": config, "data": dictionary_entries}
+        return config
 
     class Meta:
         model = Site
-        fields = ("site_data_export",)
+        fields = (
+            "config",
+            "categories",
+        )
 
 
-class CategoriesDataSerializer(serializers.ModelSerializer):
-    category = serializers.CharField(source="title")
-    parent_category = serializers.CharField(source="parent")
+class AudioDataSerializer(serializers.ModelSerializer):
+    speaker = serializers.SerializerMethodField()
+    filename = serializers.FileField(source="original.content")
+
+    @staticmethod
+    def get_speaker(audio):
+        speakers = audio.speakers.all()
+        name = speakers[0].name if speakers.count() > 0 else None
+        return name
 
     class Meta:
-        model = Category
+        model = Audio
         fields = (
-            "category",
-            "parent_category",
+            "speaker",
+            "filename",
         )
 
 
+class ImageDataSerializer(serializers.ModelSerializer):
+    filename = serializers.FileField(source="original.content")
+
+    class Meta:
+        model = Image
+        fields = ("filename",)
+
+
 class DictionaryEntryDataSerializer(serializers.ModelSerializer):
     source = serializers.SerializerMethodField()
     entryID = serializers.UUIDField(source="id")
     word = serializers.CharField(source="title")
     definition = serializers.SerializerMethodField()
     audio = serializers.SerializerMethodField()
     img = serializers.SerializerMethodField()
-    theme = CategoriesDataSerializer(source="categories", many=True)
+    theme = serializers.SerializerMethodField()
     secondary_theme = serializers.SerializerMethodField()
     optional = serializers.SerializerMethodField()
     compare_form = serializers.CharField(source="title")
     sort_form = serializers.SerializerMethodField()
     sorting_form = serializers.SerializerMethodField()
 
-    def get_source(self, dictionaryentry):
+    @staticmethod
+    def get_source(dictionaryentry):
         return dict_entry_type_mtd_conversion(dictionaryentry.type)
 
-    def get_definition(self, dictionaryentry):
-        if dictionaryentry.translation_set.first() is not None:
-            return dictionaryentry.translation_set.first().text
+    @staticmethod
+    def get_definition(dictionaryentry):
```

Comment:
Question out of curiosity, what if the dictionary entry has more that one translation? Does only the first need to be provided? If I recall correctly some of the migrated entries also had translations that were split into two for being too long, but that may only apply to longer models like songs/stories

Reply:
The MTD format appears to only specify a single definition right now but this is a good point and could be a good improvement for a future ticket.